packages/story-editor/src/components/floatingMenu/elements/layerOpacity.js

```diff
@@ -23,14 +23,30 @@ import { __ } from '@googleforcreators/i18n';
 /**
  * Internal dependencies
  */
-import { IconButton } from './shared';
+import { useStory } from '../../../app';
+import { Input, useProperties } from './shared';
 
 function LayerOpacity() {
+  const { opacity } = useProperties(['opacity']);
+  const updateSelectedElements = useStory(
+    (state) => state.actions.updateSelectedElements
+  );
+
+  const handleOpacityChange = (_, value) =>
+    updateSelectedElements({
+      properties: () => ({
+        opacity: value ?? 100,
+      }),
+    });
+
   return (
-    <IconButton
-      Icon={Icons.ColorDrop}
-      title={__('Change layer opacity', 'web-stories')}
-      onClick={() => {}}
+    <Input
+      suffix={<Icons.ColorDrop />}
+      value={opacity || 0}
```

Comment:
🤔  I think the percentage symbol should be here, even though the input space will cut it off, so that it's consistent with the other one.
`unit={_x('%', 'Percentage', 'web-stories')}`

@barklund @miina is there a future plan for an update with number inputs at all for including units that I missed?

| without | with | 
| --- | --- | 
| <img width="421" alt="Screen Shot 2022-03-10 at 5 24 28 PM" src="https://user-images.githubusercontent.com/10720454/157777957-84b1fcd4-8e7a-4825-b6d9-2014156eac04.png"> | <img width="452" alt="Screen Shot 2022-03-10 at 5 23 15 PM" src="https://user-images.githubusercontent.com/10720454/157777867-c36c4bac-08c1-4e92-a035-8968483783dc.png"> |

Reply:
I did add a placeholder ticket for another numeric input that was slightly too wide. #10799 @agingoldseco wanted to see the floating toolbars with some content before nailing down an exact size. Maybe we can address this there?

Reply:
> is there a future plan for an update with number inputs at all for including units that I missed?

As far as I know there's no separate issue for it currently, Brooke's idea sounds good.

Reply:
Sounds good, I'll leave it out for now and I can add a task to check that there